src/test/test-bpf-object.c

```diff
@@ -0,0 +1,48 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <bpf/bpf.h>
+#include <sys/resource.h>
+
+#include "bpf-object.h"
+#include "bpf/allow_bind/allow-bind-hexdump.h"
+
+#include "set.h"
+#include "tests.h"
+
+int main(int argc, char *argv[]) {
+        _cleanup_(bpf_object_freep) struct bpf_object *obj = NULL;
+        Set *progs = NULL;
+        uint8_t dummy = 1;
+        struct rlimit rl;
+        uint16_t key;
+        int r, fd;
+
+        assert_se(getrlimit(RLIMIT_MEMLOCK, &rl) >= 0);
+        rl.rlim_cur = rl.rlim_max = MAX(rl.rlim_max, CAN_MEMLOCK_SIZE);
+        (void) setrlimit(RLIMIT_MEMLOCK, &rl);
+
+        if (!can_memlock())
+                return log_tests_skipped("Can't use mlock(), skipping.");
+
+        r = bpf_object_new(allow_bind_hexdump_buffer, sizeof(allow_bind_hexdump_buffer), &obj);
```

Comment:
Is there a way to detect other limitations and skip the test gracefully? Eg, the test machine might not have BPF kconfig enabled. Bear in mind that tests can be build and ran on different machines (eg: autopkgtest in Debian/Ubuntu).

Reply:
@bluca 

Added checks for CPU architecture (x86_64 is supported) and a probe for whether bpf program type is supported.
Kernel config is a great idea, according to my grepping, systemd doesn't check it for other features, doesn't it?
I'll add it, is `src/basic/` where `parse-util` resides looks appropriate?

Reply:
I was thinking more about a runtime check than a kconfig parse. Is the presence of /sys/fs/bpf a good indicator for this?